EIPS/eip-5573.md

```diff
@@ -0,0 +1,273 @@
+
+This example is expected to transform into the following `recap-transformed-statement` (for `URI` of `https://example.com`):
+
+```text
+I further authorize did:key:example to perform the following actions on my behalf: (1) "crud": "delete", "update" for "https://example.com/pictures". (2) "other": "action" for "https://example.com/pictures". (3) "msg": "recieve", "send" for "mailto:username@example.com".
+```
+
+This example is also expected to transform into the following `recap-uri`:
+
+```text
+urn:recap:eyJhdHQiOnsiaHR0cHM6Ly9leGFtcGxlLmNvbS9waWN0dXJlcyI6eyJjcnVkL2RlbGV0ZSI6W10sImNydWQvdXBkYXRlIjpbXSwib3RoZXIvYWN0aW9uIjpbXX0sIm1haWx0bzp1c2VybmFtZUBleGFtcGxlLmNvbSI6eyJtc2cvcmVjaWV2ZSI6W3sibWF4X2NvdW50Ijo1LCJ0ZW1wbGF0ZXMiOlsibmV3c2xldHRlciIsIm1hcmtldGluZyJdfV0sIm1zZy9zZW5kIjpbeyJ0byI6ImpvZUBlbWFpbC5jb20ifV19fSwicHJmIjpbImJhZnliZWlnazdseTNwb2c2dXVweGt1M2I2YnViaXJyNDM0aWI2dGZheW12b3g2Z290YWFhYWFhYWFhIl19
+```
+
+#### ReCap Translation Algorithm
+
+After applying the ReCap Translation Algorithm on a given SIWE message that MAY include a pre-defined `statement`, the `recap-transformed-statement` in a ReCap SIWE message MUST conform to the following ABNF:
+```text
+recap-transformed-statement = statement recap-preamble 1*(" " recap-statement-entry ".")
+   ; see EIP-4361 for definition of input-statement
+recap-preamble = "I further authorize " uri " to perform the following actions on my behalf:"
+   ; see EIP-4361 for definition of uri
+recap-statement-entry = "(" number ") " action-namespace ": " 
+                          action-name *("," action-name) "for"
+                          recap-resource
+   ; see RFC8259 for definition of number
+ability-namespace = string
+   ; see RFC8259 for definition of string
+ability-name = string
+   ; see RFC8259 for definition of string
+recap-resource = string
+   ; see RFC8259 for definition of string
+```
+
+The following algorithm or an algorithm that produces the same output MUST be performed to generate the SIWE ReCap Transformed Statement.
+
+Inputs:
+- Let `uri` be the uri field of the input SIWE message conforming to EIP-4361.
+- Let `recap-uri` be a ReCap URI, which represents the ReCap Capabilities that are to be encoded in the SIWE message, and which contains a ReCap Details Object which conforms to the ReCap Details Object Schema.
+- [Optional] Let `statement` be the statement field of the input SIWE message conforming to EIP-4361.
+Algorithm:
+- Let `recap-transformed-statement` be an empty string value.
+- If `statement` is present, do the following: 
+    - Append the value of the `statement` field of `siwe` to `recap-transformed-statement`.
+    - Append a single space character `" "` to `recap-transformed-statement`.
+- Append the following string to `recap-transformed-statement`: "I further authorize ".
+- Append `uri` to `recap-transformed-statement`.
```

Comment:
As above: are we updating the statement to `stated URI` instead of the in-line URI?